apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -54,7 +54,7 @@ def get_items(
     count = 0
     for metadata in items.geojson()["features"]:
         count += 1
-        if count > limit:
+        if (limit is not None) and (count > limit):
```

Comment:
If `limit` can be None, then it should be marked as `limit: Optional[int]` rather than current `:int`